server/embed/config_test.go

```diff
@@ -40,7 +40,7 @@ func notFoundErr(service, domain string) error {
 
 func TestConfigFileOtherFields(t *testing.T) {
 	ctls := securityConfig{TrustedCAFile: "cca", CertFile: "ccert", KeyFile: "ckey"}
-	ptls := securityConfig{TrustedCAFile: "pca", CertFile: "pcert", KeyFile: "pkey"}
+	ptls := securityConfig{TrustedCAFile: "pca", CertFile: "pcert", KeyFile: "pkey", AllowedCN: "etcd"}
```

Comment:
because theoretically it is forbidden by the application logic so I thought it would be better to not introduce it in the test cases. Do you have any other thoughts on it?

Reply:
The motivation is to cover both new fields in unit test. It's OK to verify the two fields separately.

Reply:
Added.

Reply:
> It's OK to verify the two fields separately.

Sorry, I should be clearer. I was saying to verify the two fields in two separate sub-tests because they are mutually exclusive. 
https://github.com/etcd-io/etcd/blob/126e0d511f73e16b3bf87688932b7144ae04170a/client/pkg/transport/listener.go#L414-L417

[My previous suggestion](https://github.com/etcd-io/etcd/pull/17861#discussion_r1592117199) isn't appropriate, but it isn't a big deal. Please
- either add a comment something "`Note AllowedCN and AllowedHostname are mutually exclusive, this test is just to the fields can be correctly marshalled & unmarshalled`" right above the line (43).
- or add two separate sub-tests, one cover `AllowedCN`, and the other covers `AllowedHostname` 

Sorry for the back and forth.

Reply:
I put the comment as I thought it might be a good practice to track the behavior of `ConfigFromFile`.